hub/core/chunk_engine.py

```diff
@@ -558,28 +564,67 @@ def _sanitize_samples(self, samples):
             samples = list(samples)
         return samples, verified_samples
 
+    def can_fit_to_next(self, samples, fit_row):
+        chunk_id = self.chunk_id_encoder.get_next_chunk_id(fit_row)
+        if chunk_id is None:
+            return False
+
+        chunk_name = ChunkIdEncoder.name_from_id(chunk_id)  # type: ignore
+        chunk_commit_id = self.get_chunk_commit(chunk_name)
+        chunk_key = get_chunk_key(self.key, chunk_name, chunk_commit_id)
+        chunk_size = self.cache.get_object_size(chunk_key)
+
+        sample_bytes = 0
+        for _, sample in enumerate(samples):
+            sample_bytes += len(sample.buffer)
+
+        sum_size = chunk_size + sample_bytes
+        if sum_size < self.min_chunk_size:
+            return True
+        return False
+
     def _samples_to_chunks(
         self,
         samples,
         start_chunk: Optional[BaseChunk] = None,
         register: bool = True,
         update_commit_diff: bool = False,
+        append_to_end: bool = True,
+        extend: bool = True,
+        fit_row: Optional[int] = None,
     ):
+        # TODO check if there is a suboptimal chunk that we can use
```

Comment:
???

Reply:
this was here before my changes, will remove